demos/shape/heart/foundation.js

```diff
@@ -0,0 +1,121 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import MDCShapeFoundation from '../../../packages/mdc-shape/foundation';
+
+export default class DemoHeartFoundation extends MDCShapeFoundation {
+  generatePath_(width, height, padding) {
+    return 'm ' + (width / 2) + ' ' + (height - padding)
+    +this.generateLeftBase_(width, height, padding)
+    +this.generateTopLeftCurve_(width, height, padding)
+    +this.generateTopRightCurve_(width, height, padding)
+    +this.generateRightBase_(width, height, padding);
+  }
+  generateBaseWidth_(width, padding) {
+    return ((width - (padding * 2)) / 2);
+  }
+  generateBaseFlatWidth_(width, padding) {
+    return this.generateBaseWidth_(width, padding) * 0.14;
+  }
+  generateBaseCurveWidth_(width, padding) {
+    return this.generateBaseWidth_(width, padding) * 0.52;
+  }
+  generateBaseHeight_(height, padding) {
+    return (height - (padding * 2)) * 0.67;
+  }
+  generateBaseFlatHeight_(height, padding) {
+    return this.generateBaseHeight_(height, padding) * 0.11;
+  }
+  generateBaseCurveHeight_(height, padding) {
+    return this.generateBaseHeight_(height, padding) * 0.4;
+  }
+  generateBaseCurveHeight2_(height, padding) {
+    return this.generateBaseHeight_(height, padding) * 0.67;
+  }
+  generateLeftBase_(width, height, padding) {
+    const flatWidth = this.generateBaseFlatWidth_(width, padding);
+    const flatHeight = this.generateBaseFlatHeight_(height, padding);
+    const realWidth = this.generateBaseWidth_(width, padding) - flatWidth;
+    return 'l ' + (-1 * flatWidth) + ',' + (-1 * flatHeight)
+      +'c ' + (-1 * this.generateBaseCurveWidth_(width, padding)) + ','
+      + (-1 * this.generateBaseCurveHeight_(height, padding))
+      + ' ' + (-1 * realWidth) + ',' + (-1 * this.generateBaseCurveHeight2_(height, padding))
+      + ' ' + (-1 * realWidth)
+      + ',' + (-1 * (this.generateBaseHeight_(height, padding) - flatHeight));
+  }
+  generateRightBase_(width, height, padding) {
+    const flatWidth = this.generateBaseFlatWidth_(width, padding);
+    const flatHeight = this.generateBaseFlatHeight_(height, padding);
+    const curveHeight = this.generateBaseHeight_(height, padding) - flatHeight;
+    const realWidth = this.generateBaseWidth_(width, padding) - flatWidth;
+    return 'c 0,' + (curveHeight - this.generateBaseCurveHeight2_(height, padding))
+      + (-1 * (realWidth - this.generateBaseCurveWidth_(width, padding)))
+      + ',' + (curveHeight - this.generateBaseCurveHeight_(height, padding))
+      + (-1 * realWidth) + ',' + curveHeight
+      +'l ' + (-1 * flatWidth) + ',' + flatHeight;
```

Comment:
nit: space after first `+` operator.

Reply:
Another nit... do we not prefer trailing operators rather than leading? (Maybe moot if we reorganize this and use a template string.)

Reply:
My Outlined Text Field prototype uses leading operators, and the king of SVG on the web Mike Bostock always uses leading operators. I am open to more clearly defining our convention here, and am not married to leading operators.

Reply:
I guess it comes down to whether we want to adhere to [Google's JS style guide](https://google.github.io/styleguide/jsguide.html#formatting-where-to-break) for this, which promotes breaking after the operator, not before.

Reply:
I think the closer we are to the Google style guide the better. Since that is the companies encouraged policy on coding style.

However, I still think shifting these to template literals would net a simpler codebase to maintain long-term.